zulipterminal/ui_tools/boxes.py

```diff
@@ -134,10 +133,19 @@ def generic_autocomplete(self, text: str, state: Optional[int]
 
         # NOTE: The following block only executes if any of the autocomplete
         # prefixes exist.
-        self.is_in_typeahead_mode = True
         typeaheads, suggestions = (
             autocomplete_func(text[prefix_index:], prefix)
         )
+
+        typeahead = self._process_typeaheads(typeaheads, state,
+                                             suggestions)
+        if typeahead:
+            typeahead = text[:prefix_index] + typeahead
+        return typeahead
+
+    def _process_typeaheads(self, typeaheads: List[str], state: Optional[int],
```

Comment:
This is a very general method name - is there anything more specific we could use?

Reply:
I kept it general because there are multiple things happening here - returning typeahead after checking for state constraints, and setting suggestions to footer.